proto/batch_plan.proto

```diff
@@ -359,3 +359,7 @@ message PlanFragment {
   PlanNode root = 1;
   ExchangeInfo exchange_info = 2;
 }
+
+message CapturedContext {
```

Comment:
Here the `CapturedContext` message is defined in `batch_plan.proto`, while in `src/expr/core/src/captured_context.rs`, it's said to be "For all execution mode".

So shall we move that to somewhere else like `expr.proto` or `compute.proto` or `common.proto`?

Btw, please add some comments for it.

Reply:
I'm about to move it `stream_plan` as we also need this for streaming execution mode. 

For naming, what about `SessionContext`? As this struct is mainly used to capture various sessional values for various execution modes.

Reply:
I would prefer to name it `ExecutionContext` since it's about one query execution. It may include things even more limited to one execution

Reply:
`ExecutionContext` is good, but it conflicts with [an exsiting struct](https://github.com/risingwavelabs/risingwave/blob/2ddab6bc806040455c139823ed75cbcea3ef143f/src/frontend/src/scheduler/mod.rs#L47) in frontend codes. I'm concerning it will cause confusion.

Reply:
Or `ExecutionContextInfo`? In fact I expect the existing struct should contains such info.

Reply:
`Context`and `Info` together seems to be a little redundant. Maybe `CapturedExecutionContext`?

Reply:
LGTM

Reply:
> I'm about to move it stream_plan as we also need this for streaming execution mode.

Why not `plan_common.proto`?

Reply:
😂It‘s actually not used in `batch_plan.proto`, so `stream_plan.proto` is enough. But `plan_common.proto` make more sense. I'll move it to `plan_common.proto`.